src/kv/tx.h

```diff
@@ -74,13 +80,101 @@ namespace kv
       return std::make_tuple(typed_view);
     }
 
+    template <class M>
+    std::tuple<typename M::TxView*> get_tuple2(const std::string& map_name)
+    {
+      if (store == nullptr)
+      {
+        throw std::logic_error("New form called on old-style Tx");
+      }
+
+      using MapView = typename M::TxView;
+
+      // If the M is present, its AbstractTxView should be an M::TxView. This
+      // invariant could be broken by set_view_list, which will produce an error
+      // here
+      auto search = view_list.find(map_name);
+      if (search != view_list.end())
+      {
+        auto view = dynamic_cast<MapView*>(search->second.view.get());
+
+        if (view == nullptr)
+        {
+          throw std::logic_error(
+            fmt::format("View over map {} is not of expected type", map_name));
+        }
+
+        return std::make_tuple(view);
+      }
+
+      if (read_version == NoVersion)
+      {
+        // Grab opacity version that all Maps should be queried at.
+        auto txid = store->current_txid();
+        term = txid.term;
+        read_version = txid.version;
+      }
+
+      M* typed_map = nullptr;
+
+      auto type_erased_map = store->get_map(read_version, map_name);
```

Comment:
I don't understand why this variable is called `type_erased_map`?

Reply:
So its actually an `AbstractMap`. Trying to distinguish the 3 levels of typing; `AbstractMap`, `kv::untyped::Map`, and `kv::Map<K, V>`. We need one of the latter 2 to call `create_view` on. Anyway I've renamed this in a later change, so we now have variables `abstract_map`, `untyped_map`, `typed_map`.